src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -175,6 +176,25 @@ public class RemoteExecutionService {
   private final AtomicBoolean shutdown = new AtomicBoolean(false);
   private final AtomicBoolean buildInterrupted = new AtomicBoolean(false);
 
+  private boolean isActionCompatibleWithXPlatformCache(Spawn spawn) {
+    String repo = spawn.getResourceOwner().getOwner().getLabel().getPackageIdentifier().getRepository().getName();
+    if (!repo.isEmpty()) {
+      return false;
+    }
+    String mnemonic = spawn.getResourceOwner().getMnemonic();
+    return remoteOptions.remoteXPlatSupportedMnemonics.contains(mnemonic);
+  }
+
+  private boolean isInputRemovedForXPlatform(PathFragment pathFragment) {
+    String path = pathFragment.getPathString();
+    return remoteOptions.remoteXPlatRemovedInputs.stream().anyMatch(s -> path.indexOf(s) >= 0);
```

Comment:
Similar to https://github.com/bazelbuild/bazel/pull/15638 if we use regex here, it could simplify declaration when we have lot of inputs to exclude.